packages/common/infra/src/livedata/index.ts

```diff
@@ -0,0 +1,105 @@
+import { DebugLogger } from '@affine/debug';
+import type { Observable } from 'rxjs';
+import { BehaviorSubject } from 'rxjs';
+
+const logger = new DebugLogger('livedata');
+
+/**
+ * LiveData is a reactive data type.
+ *
+ * ## basic usage
+ *
+ * @example
+ * ```ts
+ * const livedata = new LiveData(0); // create livedata with initial value
+ *
+ * livedata.next(1); // update value
+ *
+ * console.log(livedata.value); // get current value
+ *
+ * livedata.subscribe(v => { // subscribe to value changes
+ *  console.log(v); // 1
+ * });
+ * ```
+ *
+ * ## observable
+ *
+ * LiveData is a rxjs observable, you can use rxjs operators.
+ *
+ * @example
+ * ```ts
+ * new LiveData(0).pipe(
+ *   map(v => v + 1),
+ *   filter(v => v > 1),
+ *   ...
+ * )
+ * ```
+ *
+ * NOTICE: different from normal observable, LiveData will always emit the latest value when you subscribe to it.
+ *
+ * ## from observable
+ *
+ * LiveData can be created from observable or from other livedata.
+ *
+ * @example
+ * ```ts
+ * const A = LiveData.from(
+ *   of(1, 2, 3, 4), // from observable
+ *   0 // initial value
+ * );
+ *
+ * const B = LiveData.from(
+ *   A.pipe(map(v => 'from a ' + v)), // from other livedata
+ *   '' // initial value
+ * );
+ * ```
+ *
+ * NOTICE: LiveData.from will not complete when the observable completes, you can use `spreadComplete` option to change
+ * this behavior.
+ *
+ * ## Why is it called LiveData
+ *
+ * This API is very similar to LiveData in Android, as both are based on Observable, so I named it LiveData.
+ *
+ * @see {@link https://rxjs.dev/api/index/class/BehaviorSubject}
+ * @see {@link https://developer.android.com/topic/libraries/architecture/livedata}
+ */
+export class LiveData<T = unknown> extends BehaviorSubject<T> {
```

Comment:
I have no knowledge on Android's LiveData at all and after reading some brief intro it seems to me that LiveData in JetPack enables lifecycle bindings to views. It seems to me that if we borrow the LiveData term, it might be used together in a hook.

Reply:
Actually, the LiveData currently implemented is not bound to any lifecycle, because the few LiveData used so far are pure, so there is no need for destruction.

If the Observable inside the LiveData has side effects, it should be unsubscribe when there are no listeners. I plan to do this later.

Reply:
Not only about observable unsubscription, the parts on data-binding for view updates looks more relevant for LiveData abstraction. Regarding side-effects, not sure if we want to directly add side effects in the piped operators - maybe we should have another abstraction over it?